internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadOnlyStoreConfigRepository.java

```diff
@@ -115,6 +121,15 @@ public Optional<StoreConfig> getStoreConfig(String storeName) {
       veniceStoreName = VeniceSystemStoreType.DAVINCI_PUSH_STATUS_STORE.extractRegularStoreName(storeName);
     }
     StoreConfig config = storeConfigMap.get().get(veniceStoreName);
+    // Lazy fetch. It should happen when the entire store config map is not loaded, and only do ad-hoc fetch
+    if (config == null) {
+      config = accessor.getStoreConfig(veniceStoreName);
+      if (config != null) {
+        storeConfigMap.get().put(config.getStoreName(), config);
+        accessor.subscribeStoreConfigDataChangedListener(veniceStoreName, storeConfigChangedListener);
+      }
+    }
+
     if (config != null) {
       return Optional.of(config.cloneStoreConfig());
     } else {
```

Comment:
There is already a branch for config being null. Shall we put the new logic here or remove it?

Reply:
After the new logic, the config can still be null or not-null so this branch still applies. We may use ternary operator to make it look nice tho